src/awkward1/highlevel.py

```diff
@@ -879,7 +879,7 @@ def __setitem__(self, where, what):
         in-place. (Internally, this method uses #ak.with_field, so performance
         is not a factor in choosing one over the other.)
         """
-        if not isinstance(where, str):
+        if not isinstance(where, (str, list, tuple)):
```

Comment:
Just str or tuple of strs here (check the contents); also the tuple can't be empty.